### PR TITLE
Support disabling migrations 

### PR DESCRIPTION
**Pull request by iantasker**
If you configure Filament to use your application users you are left with two unused tables after running `php artisan migrate`. 
It would be good to support the ability to disable migrations from running in a similar way to how Cashier allows them to be disabled.

### CODE REVIEW

**Comment by ryangjchandler** [comment]
@iantasker There are some merge conflicts with `develop` which is where all PRs get merged into to. Would you be able to either update your fork based on the origin, or re-create your pull request from `develop`, or just fix the conflicts yourself.

File: src/FilamentManager.php [hunk 1]
```diff
@@ -64,4 +71,26 @@ public function userResource()
...
+
+        return new static;
+    }
+
+    /**
+     * Determine if Filament should run its migrations.
+     *
+     * @return bool
+     */
+    public static function runsMigrations()
```
Comment (ryangjchandler) [inline src/FilamentManager.php:92]:
```suggestion
    public function shouldRunMigrations()
```

File: src/FilamentManager.php [hunk 2]
```diff
@@ -64,4 +71,26 @@ public function userResource()
...
+    {
+        return static::$runsMigrations;
```
Comment (ryangjchandler) [inline src/FilamentManager.php:94]:
```suggestion
        return static::$shouldRunMigrations;
```

File: src/FilamentManager.php [hunk 3]
```diff
@@ -64,4 +71,26 @@ public function userResource()
     {
         return config('filament.user_resource', UserResource::class);
     }
+
+    /**
+     * Configure Filament to not register its migrations.
+     *
+     * @return static
+     */
+    public static function ignoreMigrations()
```
Comment (ryangjchandler) [inline src/FilamentManager.php:80]:
```suggestion
    public function ignoreMigrations()
```

File: src/FilamentManager.php [hunk 4]
```diff
@@ -64,4 +71,26 @@ public function userResource()
...
+    {
+        static::$runsMigrations = false;
```
Comment (ryangjchandler) [inline src/FilamentManager.php:82]:
```suggestion
        static::$shouldRunMigrations = false;
```

File: src/FilamentManager.php [hunk 5]
```diff
@@ -64,4 +71,26 @@ public function userResource()
...
+
+        return new static;
```
Comment (ryangjchandler) [inline src/FilamentManager.php:84]:
```suggestion

```

File: src/FilamentManager.php [hunk 6]
```diff
@@ -64,4 +71,26 @@ public function userResource()
     {
         return config('filament.user_resource', UserResource::class);
     }
+
+    /**
```
Comment (ryangjchandler) [inline src/FilamentManager.php:75]: No need for the DocBlock.

File: src/FilamentManager.php [hunk 7]
```diff
@@ -15,6 +15,13 @@ class FilamentManager
 
     public $widgets = [];
 
+    /**
+     * Indicates if Filament migrations will be run.
+     *
+     * @var bool
+     */
+    public static $runsMigrations = true;
```
Comment (ryangjchandler) [inline src/FilamentManager.php:23]:
```suggestion
    public $shouldRunMigrations = true;
```

File: src/FilamentManager.php [hunk 8]
```diff
@@ -15,6 +15,13 @@ class FilamentManager
 
     public $widgets = [];
 
+    /**
```
Comment (ryangjchandler) [inline src/FilamentManager.php:18]: No need for the DocBlock.

File: src/Filament.php [hunk 9]
```diff
@@ -17,6 +17,8 @@
  * @method static void registerRole(string $role)
  * @method static void registerWidget(string $widget)
  * @method static Resource userResource()
+ * @method static void ignoreMigrations()
+ * @method static Boolean runsMigrations()
```
Comment (ryangjchandler) [inline src/Filament.php:21]:
```suggestion
 * @method static bool shouldRunMigrations()
```

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

File: src/FilamentManager.php (hunk 1 above)
Comment (iantasker) [reply src/FilamentManager.php:204]: @ryangjchandler I believe this should remain a static function so it can be called via the facade.

**Comment by iantasker** [comment]
@ryangjchandler I have resolved all the conflicts. I will remind myself to submit PRs against develop in future.

File: src/FilamentManager.php [hunk 10]
```diff
@@ -193,4 +195,14 @@ public function version()
 
         return InstalledVersions::getPrettyVersion('filament/filament');
     }
+
+    public static function ignoreMigrations()
+    {
+        static::$shouldRunMigrations = false;
+    }
+
+    public static function runsMigrations()
```
Comment (ryangjchandler) [inline src/FilamentManager.php:204]:
```suggestion
    public function shouldRunMigrations()
```

This is a naming convention used elsewhere in Filament so we should keep it consistent.

File: src/FilamentManager.php [hunk 11]
```diff
@@ -193,4 +195,14 @@ public function version()
 
         return InstalledVersions::getPrettyVersion('filament/filament');
     }
+
+    public static function ignoreMigrations()
```
Comment (ryangjchandler) [inline src/FilamentManager.php:199]:
```suggestion
    public function ignoreMigrations()
```

Everything is being proxied through the facade, so no need to make this static.

File: src/FilamentManager.php [hunk 12]
```diff
@@ -27,6 +27,8 @@ class FilamentManager
 
     public $widgets = [];
 
+    public static $shouldRunMigrations = true;
```
Comment (ryangjchandler) [inline src/FilamentManager.php:30]:
```suggestion
    public $shouldRunMigrations = true;
```

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

**Comment by iantasker** [comment]
@ryangjchandler I have updated to match your naming convention.

File: src/FilamentManager.php [hunk 13]
```diff
@@ -196,13 +196,13 @@ public function version()
...
 
-    public static function ignoreMigrations()
+    public function ignoreMigrations()
     {
-        static::$shouldRunMigrations = false;
+        $this->shouldRunMigrations = false;
     }
 
-    public static function runsMigrations()
+    public function runsMigrations()
```
Comment (ryangjchandler) [inline src/FilamentManager.php:204]:
```suggestion
    public function shouldRunMigrations()
```

Change the method name here - it's a naming convention used elsewhere in Filament.

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

**Review by ryangjchandler — APPROVED** [review]

**Comment by danharrin** [comment]
Thanks Ian!

**Comment by iantasker** [comment]
@danharrin Your welcome, any ETA on when a new version will be released?

**Comment by danharrin** [comment]
> @danharrin Your welcome, any ETA on when a new version will be released?

Very soon, we have a couple of large features merged so the next release will be minor.

**Pull request merged**